app/models/user_notification.rb

```diff
@@ -25,4 +25,24 @@ class UserNotification < ActiveRecord::Base
   # Associations
   belongs_to :user
   belongs_to :notification
+
+  def generate_token
+    return token if token.present?
+
+    user_notifications = UserNotification.where(user_id: user_id)
+
+    if user_notifications.any? { |u_n| u_n.token.present? }
```

Comment:
Why do you want the `token` to be the same across user notifications with the same user?